components/omega/doc/design/PGrad.md

```diff
@@ -0,0 +1,196 @@
+
+### 2.4 Requirement: Flexibility to support tidal forcing and sea level change
+In later versions of Omega, the pressure gradient will need to be able to include tidal forcing in the geopotential term.
+These tidal forcings include both the tidal potential and the self attraction and loading terms.
+Additionally, other changes to the geoid
+
+### 2.5 Requirement: Pressure gradient for barotropic mode
+
+For split barotropic-baroclinic timestepping, the pressure gradient should provide the bottom pressure gradient tendency in the barotropic mode.
+
+### Desired:
+
+## 3 Algorithmic Formulation
+The non-Boussinesq momentum equation is
+$$ \frac{D \mathbf{u}_h}{D t } + f\boldsymbol{k}\times \mathbf{u}_h + \left(v\nabla_A p + \nabla_A \phi \right) = \boldsymbol{\mathcal{F}}. $$
+
+where $\mathbf{u}_h$ is the horizontal velocity, $f$ is the Coriolis parameter, $v = \frac{1}{\rho}$ is the specific volume, $\rho = \rho(T,S,p)$ is the density, $p$ is the hydrostatic pressure, $\phi$ is the geopotential, and $\boldsymbol{\mathcal{F}}$ are the dissipative terms.
+The operator $\nabla_A$ is the gradient along a constant surface, $A$, and the total derivative is
+
+$$ \frac{D \mathbf{u}_h}{D t } = \left( \frac{\partial}{\partial t} \right)_A  + \mathbf{u}_h\cdot \nabla_A + \omega\frac{\partial}{\partial A}, $$
+
+where $\omega$ is the cross coordinate flow.
+In the layered non-Boussinesq equations, the prognostic variable is the pressure thickness $h_k$, so that the geometric thickness (in meters) is a diagnostic variable defined as:
+
+$$ \Delta z_k = v_k h_k. $$
+
+The pressure at vertical cell interfaces is the found by summing the pressure thicknesses:
+
+$$ p_{K+1/2} = p_{surf} + g\sum_{k=1}^K h_k. $$
+
+The geopotential at vertical cell integrates is found by summing the pressure thicknesses multiplied by the specific volume:
+
+$$ \phi_{K+1/2} = g\left(z_b + \sum_{k=K}^{N} v_k h_k\right), $$
+
+where $z_b$ is the (positive-up) bottom depth.
+
+The discrete gradient operator at an edge is:
+
+$$ \nabla {(\cdot)} = \frac{1}{d_e} \sum_{i\in CE(e)} -n_{e,i} (\cdot)_i $$
+
+where $d_e$ is the distance between cell centers, $CE(e)$ are the cells on edge $e$, and $n_{e,i}$ is the sign of the edge normal with respect to cell $i$.
+Therefore the centered pressure gradient will be calculated as:
+
+$$ T^p_k = \frac{1}{d_e} \left( \widehat{v}_{k,e} \sum_{i \in CE(e)} n_{e,i} \overline{p}_{k,i} + \sum_{i\in CE(e)} n_{e,i} \overline{\phi}_{k,i}\right), $$
+
+$$ = \frac{1}{d_e} \left(  \sum_{i \in CE(e)} n_{e,i} \left( \widehat{v}_{k,e}\overline{p}_{k,i} + \overline{\phi}_{k,i} \right) \right), $$
+
+with the vertical averaging operator defined as:
+
+$$ \overline{(\cdot)} = \frac{1}{2} \left((\cdot)_{k+1/2} + (\cdot)_{k-1/2} \right)$$
+
+and the horizontal averaging operator:
+
+$$\widehat{(\cdot)} = \frac{1}{2} \left( (\cdot)_{i=1} + (\cdot)_{i=2}\right)$$
```

Comment:
For uniformity, see notation in [Discrete Equations](https://portal.nersc.gov/project/e3sm/mpeterse/omega/design-doc-v1-eqns/_build/html/design/OmegaV1GoverningEqns.html#discrete-equations). Each variable has a native location (cell, edge, vertex) denoted by subscript (i,e,v). I used square brackets with a subscript to denote a horizontal average, like this:
![image](https://github.com/user-attachments/assets/dddfbba1-97bd-4bc8-bd7a-82b6d776dae8)
These are defined with that notation in the [V0 doc](https://docs.e3sm.org/Omega/omega/develop/design/OmegaV0ShallowWater.html#cell-to-edge-interpolation).

Reply:
Also, the $i$ in your equation are not specified. The better notation is 
![image](https://github.com/user-attachments/assets/c8aa59e2-2bee-4f11-8b46-42f5ff74ca5a)